src/printer.cpp

```diff
@@ -133,17 +133,24 @@ std::string printConnections(const ComponentMap &componentMap, const VariableMap
 
 std::string Printer::PrinterImpl::printMath(const std::string &math)
 {
+    static const std::string wrapElementName = "math_wrap_as_single_root_element";
     static const std::regex before(">[\\s\n\t]*");
     static const std::regex after("[\\s\n\t]*<");
     static const std::regex xmlDeclaration(R"|(<\?xml[[:space:]]+version=.*\?>)|");
 
     XmlDocPtr xmlDoc = std::make_shared<XmlDoc>();
     xmlKeepBlanksDefault(0);
-    xmlDoc->parse(math);
+    // Remove any XML declarations from the string.
+    std::string normalisedMath = std::regex_replace(math, xmlDeclaration, "");
+    xmlDoc->parse("<" + wrapElementName + ">" + normalisedMath + "</" + wrapElementName + ">");
```

Comment:
Wrapping the math string in an element like this actually improves the behaviour of libcellml as shown in `TEST(Model, cleanEmptyComponents)` - where content that used to get silently thrown out as it wasn't a valid XML string (i.e., placeholder text) is now correctly maintained.

(just noting this while I remember so that I might find it for future reference!)